sim/dex.ts

```diff
@@ -132,6 +134,60 @@ const Natures: {[k: string]: Nature} = {
 };
 export const toID = Data.toID;
 
+// interface for the builder.
+interface FormatSection {
+	section: string;
+	column?: number;
+	formats: FormatData[];
+}
+
+// function for merging the two lists
+function merge(main: FormatList[], custom: FormatList[]): FormatList[] {
+	// result that is return and makes the actual list for formats.
+	const result: FormatList[] = [];
+
+	// used as a intermediary to build the final list.
+	const build: FormatSection[] = [];
+
+	// used to track current section to keep formats under their sections.
+	let current: FormatSection | undefiend = {section: "", formats: []};
```

Comment:
```suggestion
	let current: FormatSection | undefined = {section: "", formats: []};
```

Reply:
Ugh, I forgot that it was temporarily undefined at one point. I should have used a temporary for that result. Sorry about that.